nifi-nar-bundles/nifi-standard-services/nifi-distributed-cache-services-bundle/nifi-distributed-cache-client-service/src/main/java/org/apache/nifi/distributed/cache/client/CacheClientHandshakeHandler.java

```diff
@@ -64,24 +65,36 @@ public class CacheClientHandshakeHandler extends ChannelInboundHandlerAdapter {
      */
     private final VersionNegotiator versionNegotiator;
 
+    /**
+     * THe network timeout associated with handshake completion
+     */
+    private final long timeoutMillis;
+
     /**
      * Constructor.
      *
      * @param channel           the channel to which this {@link io.netty.channel.ChannelHandler} is bound.
      * @param versionNegotiator coordinator used to broker the version of the distributed cache protocol with the service
+     * @param timeoutMillis     the network timeout associated with handshake completion
      */
-    public CacheClientHandshakeHandler(final Channel channel, final VersionNegotiator versionNegotiator) {
+    public CacheClientHandshakeHandler(final Channel channel, final VersionNegotiator versionNegotiator,
+                                       final long timeoutMillis) {
         this.promiseHandshakeComplete = channel.newPromise();
         this.protocol = new AtomicInteger(PROTOCOL_UNINITIALIZED);
         this.versionNegotiator = versionNegotiator;
+        this.timeoutMillis = timeoutMillis;
     }
 
     /**
      * API providing client application with visibility into the handshake process.  Distributed cache requests
      * should not be sent using this {@link Channel} until the handshake is complete.
      */
     public void waitHandshakeComplete() {
```

Comment:
maybe add that isSuccess() should be called following completion of this method (i.e. it is only safe to continue processing ifSuccess() is true)